com.vrcfury.vrcfury/SPS/sps_main.cginc

```diff
@@ -58,18 +58,34 @@ void sps_apply_real(inout float3 vertex, inout float3 normal, uint vertexId, ino
 			applyLerp = min(applyLerp, 1-entranceAngleTooSharp);
 		}
 		
+		const float hiltedSphereRadius = 0.3;
 		if (!isRing) {
 			// Uncancel if hilted in a hole
-			const float hiltedSphereRadius = 0.3;
 			const float inSphere = orfDistance > worldLength*hiltedSphereRadius ? 0 : 1;
 			//const float hilted = min(isBehind, inSphere);
 			//shrinkLerp = hilted;
 			const float hilted = inSphere;
 			applyLerp = max(applyLerp, hilted);
 		} else {
-			// Cancel if ring is near or behind base
-			const float isBehind = rootPos.z > 0 ? 0 : 1;
-			applyLerp = min(applyLerp, 1-isBehind);
+			// Align to axis if hilted in a ring
+			// Prevent orifice from getting too close to plug as that can introduce excessive roll
+			const float minDistance = worldLength * 0.2;
+			const float lerpDistance = minDistance;
+			
+			// Consider plane aligned to orifice
+			const float orfPerpDistance = dot(sps_normalize(-frontNormal), rootPos);
+			float shiftLerp = (orfPerpDistance > minDistance) | (orfPerpDistance < -worldLength * hiltedSphereRadius) ? 0 : 1;
+			
+			const float shiftAmount = (minDistance - orfPerpDistance);
+			const float3 shiftedRoot = rootPos - sps_normalize(frontNormal) * shiftAmount;
```

Comment:
this arbitrarily just moves the socket backward, in respect to the direction it is facing, which I also don't think is correct in this scenario

Reply:
The idea was based on your suggestion to "pretend" the rings are a bit forward.

Where you thinking of a different method when you suggested that?

Reply:
A Sketch showing what I'm attempting to achieve
![Drawing2 p1_cr](https://github.com/VRCFury/VRCFury/assets/4604733/8666a5a7-e853-4559-a31f-1dbe54e2b40f)